state/StateGetPointerToDataTemplate.H

```diff
@@ -0,0 +1,62 @@
+#ifdef DIMENSIONS_
+#undef DIMENSIONS_
+#endif
+
+#include "overload.macro"
+
+subroutine SUB_(state, ptr, name, not_found_ok, rc)
+   type(ESMF_State), intent(INOUT) :: state
+   real(kind=EKIND_), pointer :: ptr DIMENSIONS_
+   character(len=*), intent(IN) :: name
+   logical, optional,intent(IN) :: not_found_ok
```

Comment:
I realize that this is probably how this argument is spelled in MAP2.  But `is_present` (or ESMF's `isPresent`) is more canonical.

Changes things a bit.  `is_present` is intent(out), while `not_found_ok` is intent(in).   So need to be careful in changing use cases.  Should be very rare at least.

Reply:
Replace `intent(in)` arg `not_found_ok` with `optional, intent(out)` arg `is_present`